src/sentry/sentry_apps/models/sentry_app.py

```diff
@@ -34,25 +35,72 @@
 from sentry.types.region import find_all_region_names, find_regions_for_sentry_app
 from sentry.utils import metrics
 
+
+class SentryAppResourceType(StrEnum):
+    ISSUE = "issue"
+    ERROR = "error"
+    COMMENT = "comment"
+    INSTALLATION = "installation"
+    METRIC_ALERT = "metric_alert"
+
+    # Represents an issue alert resource
+    EVENT_ALERT = "event_alert"
+
+
+class SentryAppActionType(StrEnum):
+    # Generic actions
+    CREATED = "created"
+    DELETED = "deleted"
+    UPDATED = "updated"
+
+    # Issue actions
+    ASSIGNED = "assigned"
+    IGNORED = "ignored"
+    RESOLVED = "resolved"
+    UNRESOLVED = "unresolved"
+
+    # Issue alert actions
+    TRIGGERED = "triggered"
+
+    # Metric alert actions
+    OPEN = "open"
+    CRITICAL = "critical"
+    WARNING = "warning"
+
+
+# Define event type unions for each resource
+IssueEventType = Literal[
+    "issue.assigned", "issue.created", "issue.ignored", "issue.resolved", "issue.unresolved"
+]
```

Comment:
On a similar note, is there a way to map a single `SentryAppResourceType` to its valid set of action types instead of redefining both?. For some inspiration, check out our outbox enum handling code: https://github.com/getsentry/sentry/blob/4fbecea3d3b2b0ccca9b81390190c631516b2455/src/sentry/hybridcloud/outbox/category.py#L249

It's not perfect, but it reduces some of this boilerplate at least.